src/integer_mod_q/mat_zq/get.rs

```diff
@@ -0,0 +1,275 @@
+    /// if the number of rows or columns is greater than the matrix or negative.
+    fn get_entry(
+        &self,
+        row: impl TryInto<i64> + Display + Copy,
+        column: impl TryInto<i64> + Display + Copy,
+    ) -> Result<Zq, MathError> {
+        let (row_i64, column_i64) = evaluate_coordinates(self, row, column)?;
+
+        let value = unsafe { *fmpz_mod_mat_entry(&self.matrix, row_i64, column_i64) };
+        let modulus = self.get_mod()?;
+
+        Ok(Zq::from_z_modulus(&Z { value }, &modulus))
+    }
+}
+
+#[cfg(test)]
+mod test_get_entry {
+    use super::Zq;
+    use crate::{
+        error::MathError,
+        integer::Z,
+        integer_mod_q::MatZq,
+        traits::{GetEntry, SetEntry},
+    };
+    use std::str::FromStr;
+
+    /// Ensure that getting entries works with large numbers.
+    #[test]
+    fn max_int_positive() {
+        let mut matrix = MatZq::new(5, 10, u64::MAX).unwrap();
+        let value = Z::from(i64::MAX);
+        matrix.set_entry(0, 0, value).unwrap();
+
+        let entry = matrix.get_entry(0, 0).unwrap();
+
+        assert_eq!(Z::from(i64::MAX), entry);
+    }
+
+    /// Ensure that getting entries works with large numerators and denominators (larger than [`i64`]).
+    #[test]
+    fn big_positive() {
+        let mut matrix = MatZq::new(5, 10, u64::MAX).unwrap();
+        let value = Z::from(u64::MAX - 1);
+        matrix.set_entry(0, 0, value).unwrap();
+
+        let entry = matrix.get_entry(0, 0).unwrap();
+
+        assert_eq!(Z::from(u64::MAX - 1), entry);
+    }
+
+    /// Ensure that getting entries works with large numbers.
+    #[test]
+    fn max_int_negative() {
+        let mut matrix = MatZq::new(5, 10, u64::MAX).unwrap();
+        let value = Z::from(-i64::MAX);
+        matrix.set_entry(0, 0, value).unwrap();
+
+        let entry = matrix.get_entry(0, 0).unwrap();
+
+        assert_eq!(Z::from((u64::MAX as i128 - i64::MAX as i128) as u64), entry);
+    }
+
+    /// Ensure that getting entries works with large numerators and denominators (larger than [`i64`]).
+    #[test]
+    fn big_negative() {
+        let mut matrix = MatZq::new(5, 10, u64::MAX).unwrap();
+        let value = Z::from(-i64::MAX - 1);
+        matrix.set_entry(0, 0, value).unwrap();
+
+        let entry = matrix.get_entry(0, 0).unwrap();
+
+        assert_eq!(
+            Z::from((u64::MAX as i128 - i64::MAX as i128) as u64 - 1),
+            entry
+        );
+    }
+
+    /// Ensure that a wrong number of rows yields an Error.
+    #[test]
+    fn error_wrong_row() {
+        let matrix = MatZq::new(5, 10, 7).unwrap();
+        let entry: Result<Zq, MathError> = matrix.get_entry(5, 1);
+
+        assert!(entry.is_err());
+    }
+
+    /// Ensure that a wrong number of columns yields an Error.
+    #[test]
+    fn error_wrong_column() {
+        let matrix = MatZq::new(5, 10, 7).unwrap();
+        let entry: Result<Zq, MathError> = matrix.get_entry(1, 100);
```

Comment:
Please also test, if you are directly on the edge, i.e. a coordinate such as (5, 10)